Serve "scene.html" as default

@michaltakac yep that's what I did, it works, but I'm having problems to serve `tmp/dcl-app` when `isDev` is `true`. It looks like it's trying to require the directory?

```js
const baseDir = isDev ? './tmp/dcl-app/' : './'
  budo(baseDir, {
    host: '0.0.0.0',
    live: true,
    port: 2044,
    stream: process.stdout,
    staticOptions: {
      index: 'scene.html'
    }
  });
```

```
Cannot find module '/Users/juancazala/Code/GitHub/cli/tmp/dcl-app'
```

It works fine if I leave the `baseDir` as `'.'`

--- a/src/commands/info.ts
+++ b/src/commands/info.ts
@@ -1,6 +1,6 @@
 import { wrapCommand } from '../utils/wrapCommand'
 import { formatDictionary, italic } from '../utils/logging'
-import { Decentraland } from '../lib/Decentraland'
+import { Decentraland, Estate } from '../lib/Decentraland'
 import * as Coordinates from '../utils/coordinateHelpers'
 import { Analytics } from '../utils/analytics'
 
@@ -33,6 +33,8 @@ export function info(vorpal: any) {
         return `info coord:${target} ${rest}`
       } else if (target.startsWith('0x')) {
         return `info address:${target} ${rest}`
+      } else if (Number.isInteger(parseInt(target, 10))) {
+        return `info estate:${target} ${rest}`
       }
       return command
     })
@@ -42,18 +44,25 @@ export function info(vorpal: any) {
 
         if (!args.target) {
           await dcl.project.validateExistingProject()
+          const estateId = await dcl.project.getEstate()
+          if (estateId) {
+            Analytics.infoCmd({ type: 'estate', target: estateId })
+            const estate = await dcl.getEstateInfo(estateId)
+            return logEstate(vorpal, estate, estateId)
+          }
           const coords = await dcl.project.getParcelCoordinates()
-          const scene = await dcl.getProjectInfo(coords.x, coords.y)
           Analytics.infoCmd({ type: 'coordinates', target: coords })
-          logInfo(vorpal, scene)
-        } else if (args.target.startsWith('address:')) {
+          const scene = await dcl.getProjectInfo(coords.x, coords.y)
+          return logInfo(vorpal, scene)
+        }
+
+        if (args.target.startsWith('address:')) {
           const address = args.target.replace('address:', '')
+          Analytics.infoCmd({ type: 'address', target: address })
           const parcels = await dcl.getAddressInfo(address)
           const formatted = parcels.reduce((acc, parcel) => {
             return { ...acc, [`${parcel.x},${parcel.y}`]: { name: parcel.name, description: parcel.description, ipns: parcel.ipns } }
           }, {})
-
-          Analytics.infoCmd({ type: 'address', target: address })
 
           if (parcels.length === 0) {
             vorpal.log(italic('\n  No information available\n'))
@@ -61,16 +70,26 @@ export function info(vorpal: any) {
             vorpal.log(`\n  LAND owned by ${address}:\n`)
             vorpal.log(formatDictionary(formatted, { spacing: 2, padding: 2 }))
           }
-        } else if (args.target.startsWith('coord:')) {
+          return
+        }
+
+        if (args.target.startsWith('coord:')) {
           const raw = args.target.replace('coord:', '')
           const coords = Coordinates.getObject(raw)
-          const scene = await dcl.getParcelInfo(coords.x, coords.y)
           Analytics.infoCmd({ type: 'coordinates', target: coords })
-          logInfo(vorpal, scene)
-        } else {
-          vorpal.log(`\n  Invalid argument: ${args.target}`)
-          vorpal.log(example)
+          const data = await dcl.getParcelInfo(coords.x, coords.y)
+          return logInfo(vorpal, data)
         }
+
+        if (args.target.startsWith('estate:')) {
+          const estateId = parseInt(args.target.replace('estate:', ''), 10)
+          Analytics.infoCmd({ type: 'estate', target: estateId })
+          const estate = await dcl.getEstateInfo(estateId)
+          return logEstate(vorpal, estate, estateId)
+        }
+
+        vorpal.log(`\n  Invalid argument: ${args.target}`)
+        vorpal.log(example)
       })
     )
 }
@@ -90,5 +109,18 @@ export function logInfo(vorpal, scene) {
     vorpal.log(formatDictionary(scene.land, { spacing: 2, padding: 2 }))
   } else {
     vorpal.log(italic('    No information available\n'))
+  }
+}
+
+export function logEstate(vorpal, estate: Estate, id: number) {
+  if (!estate) {
+    vorpal.log(italic(`\n  Estate with ID ${id} doesn't exist\n`))
+    return
+  }
+
+  vorpal.log('  Estate Metadata:\n')
+
+  if (estate) {
+    vorpal.log(formatDictionary(estate, { spacing: 2, padding: 2 }))
   }
 }

--- a/src/commands/info.ts
+++ b/src/commands/info.ts
@@ -1,8 +1,8 @@
 import { wrapCommand } from '../utils/wrapCommand'
 import { formatDictionary, italic } from '../utils/logging'
 import { Decentraland, Estate } from '../lib/Decentraland'
-import * as Coordinates from '../utils/coordinateHelpers'
 import { Analytics } from '../utils/analytics'
+import { Coords, getObject, isValid } from '../utils/coordinateHelpers'
 
 export interface IArguments {
   target: string
@@ -16,6 +16,7 @@ const example = `
     dcl info      - Returns the information of the current project
     dcl info 1,1  - Returns the information of the LAND located at the given coordinates
     dcl info 0x.. - Returns the information of all LANDs owned by the specified Ethereum address
+    dcl info 5    - Returns the information of the Estate with the given ID
 `
 
 export function info(vorpal: any) {
@@ -29,7 +30,7 @@ export function info(vorpal: any) {
       const target = parts.shift()
       const rest = parts.join(' ')
 
-      if (Coordinates.isValid(target)) {
+      if (isValid(target)) {
         return `info coord:${target} ${rest}`
       } else if (target.startsWith('0x')) {
         return `info address:${target} ${rest}`
@@ -46,52 +47,69 @@ export function info(vorpal: any) {
           await dcl.project.validateExistingProject()
           const estateId = await dcl.project.getEstate()
           if (estateId) {
-            Analytics.infoCmd({ type: 'estate', target: estateId })
-            const estate = await dcl.getEstateInfo(estateId)
-            return logEstate(vorpal, estate, estateId)
+            return infoEstate(vorpal, dcl, estateId)
           }
           const coords = await dcl.project.getParcelCoordinates()
-          Analytics.infoCmd({ type: 'coordinates', target: coords })
-          const scene = await dcl.getProjectInfo(coords.x, coords.y)
-          return logInfo(vorpal, scene)
+          return infoParcel(vorpal, dcl, coords)
         }
 
         if (args.target.startsWith('address:')) {
           const address = args.target.replace('address:', '')
           Analytics.infoCmd({ type: 'address', target: address })
-          const parcels = await dcl.getAddressInfo(address)
-          const formatted = parcels.reduce((acc, parcel) => {
+          const { parcels, estates } = await dcl.getAddressInfo(address)
+
+          const formattedParcels = parcels.reduce((acc, parcel) => {
             return { ...acc, [`${parcel.x},${parcel.y}`]: { name: parcel.name, description: parcel.description, ipns: parcel.ipns } }
           }, {})
+          const formattedEstates = estates.reduce((acc, estate) => {
+            return { ...acc, [`ID ${estate.id.toString()}`]: { name: estate.name, description: estate.description, ipns: estate.ipns } }
+          }, {})
 
-          if (parcels.length === 0) {
-            vorpal.log(italic('\n  No information available\n'))
-          } else {
-            vorpal.log(`\n  LAND owned by ${address}:\n`)
-            vorpal.log(formatDictionary(formatted, { spacing: 2, padding: 2 }))
+          if (parcels.length === 0 && estates.length === 0) {
+            return vorpal.log(italic('\n  No information available\n'))
           }
+
+          if (parcels.length !== 0) {
+            vorpal.log(`\n  LAND owned by ${address}:\n`)
+            vorpal.log(formatDictionary(formattedParcels, { spacing: 2, padding: 2 }))
+          }
+
+          if (estates.length !== 0) {
+            vorpal.log(`\n  Estates owned by ${address}:\n`)
+            vorpal.log(formatDictionary(formattedEstates, { spacing: 2, padding: 2 }))
+          }
+
           return
         }
 
         if (args.target.startsWith('coord:')) {
           const raw = args.target.replace('coord:', '')
-          const coords = Coordinates.getObject(raw)
-          Analytics.infoCmd({ type: 'coordinates', target: coords })
-          const data = await dcl.getParcelInfo(coords.x, coords.y)
-          return logInfo(vorpal, data)
+          const coords = getObject(raw)
+          return infoParcel(vorpal, dcl, coords)
         }
 
         if (args.target.startsWith('estate:')) {
           const estateId = parseInt(args.target.replace('estate:', ''), 10)
-          Analytics.infoCmd({ type: 'estate', target: estateId })
-          const estate = await dcl.getEstateInfo(estateId)
-          return logEstate(vorpal, estate, estateId)
+          return infoEstate(vorpal, dcl, estateId)
         }
 
         vorpal.log(`\n  Invalid argument: ${args.target}`)
         vorpal.log(example)
       })
     )
+}
+
+async function infoParcel(vorpal, dcl: Decentraland, coords: Coords) {
+  Analytics.infoCmd({ type: 'coordinates', target: coords })
+  const estate = await dcl.getEstateOfParcel(coords)
+  const data = await dcl.getParcelInfo(coords)
+  return logInfo(vorpal, estate ? { ...data, estate } : data)
+}
+
+async function infoEstate(vorpal, dcl: Decentraland, estateId: number) {
+  Analytics.infoCmd({ type: 'estate', target: estateId })
+  const estate = await dcl.getEstateInfo(estateId)
+  return logEstate(vorpal, estate, estateId)
 }
 
 export function logInfo(vorpal, scene) {
@@ -109,6 +127,10 @@ export function logInfo(vorpal, scene) {
     vorpal.log(formatDictionary(scene.land, { spacing: 2, padding: 2 }))
   } else {
     vorpal.log(italic('    No information available\n'))
+  }
+
+  if (scene.estate) {
+    logEstate(vorpal, scene.estate, null)
   }
 }
 

--- a/src/lib/Decentraland.ts
+++ b/src/lib/Decentraland.ts
@@ -1,12 +1,14 @@
 import { EventEmitter } from 'events'
+import * as events from 'wildcards'
+
 import { IPFS, IResolveDependency } from './IPFS'
 import { Project, BoilerplateType, IFile } from './Project'
-import { Ethereum } from './Ethereum'
-import * as events from 'wildcards'
-import { getRootPath } from '../utils/project'
+import { Ethereum, LANDData } from './Ethereum'
 import { LinkerAPI } from './LinkerAPI'
 import { Preview } from './Preview'
+import { getRootPath } from '../utils/project'
 import { ErrorType, fail } from '../utils/errors'
+import { ICoords } from '../utils/coordinateHelpers'
 
 export interface IDecentralandArguments {
   workingDir?: string
@@ -24,6 +26,20 @@ export interface IAddressInfo {
   name: string
   description: string
   ipns: string
+}
+
+export type Parcel = LANDData & {
+  owner: string
+  operators?: string[]
+}
+
+export type Estate = Parcel & {
+  parcels: ICoords[]
+}
+
+export type ParcelMetadata = {
+  scene: DCL.SceneMetadata
+  land: Parcel
 }
 
 export class Decentraland extends EventEmitter {
@@ -146,11 +162,22 @@ export class Decentraland extends EventEmitter {
     return { scene, land: { ...land, owner } }
   }
 
-  async getParcelInfo(x: number, y: number) {
+  async getParcelInfo(x: number, y: number): Promise<ParcelMetadata> {
     const scene = await this.localIPFS.getRemoteSceneMetadata(x, y)
     const land = await this.ethereum.getLandData(x, y)
     const owner = await this.ethereum.getLandOwner(x, y)
     return { scene, land: { ...land, owner } }
+  }
+
+  async getEstateInfo(estateId: number): Promise<Estate> {
+    const estate = await this.ethereum.getEstateData(estateId)
+    if (!estate) {
+      return
+    }
+
+    const owner = await this.ethereum.getEstateOwner(estateId)
+    const parcels = await this.ethereum.getLandOfEstate(estateId)
+    return { ...estate, owner, parcels }
   }
 
   async getParcelStatus(x: number, y: number): Promise<{ lastModified?: string; files: IResolveDependency[] }> {

--- a/src/lib/Decentraland.ts
+++ b/src/lib/Decentraland.ts
@@ -8,7 +8,7 @@ import { LinkerAPI } from './LinkerAPI'
 import { Preview } from './Preview'
 import { getRootPath } from '../utils/project'
 import { ErrorType, fail } from '../utils/errors'
-import { ICoords } from '../utils/coordinateHelpers'
+import { Coords } from '../utils/coordinateHelpers'
 
 export interface IDecentralandArguments {
   workingDir?: string
@@ -20,13 +20,7 @@ export interface IDecentralandArguments {
   watch?: boolean
 }
 
-export interface IAddressInfo {
-  x: number
-  y: number
-  name: string
-  description: string
-  ipns: string
-}
+export type AddressInfo = { parcels: ({ x: number; y: number } & LANDData)[]; estates: ({ id: number } & LANDData)[] }
 
 export type Parcel = LANDData & {
   owner: string
@@ -34,7 +28,7 @@ export type Parcel = LANDData & {
 }
 
 export type Estate = Parcel & {
-  parcels: ICoords[]
+  parcels: Coords[]
 }
 
 export type ParcelMetadata = {
@@ -141,14 +135,32 @@ export class Decentraland extends EventEmitter {
     await preview.startServer(this.options.previewPort)
   }
 
-  async getAddressInfo(address: string): Promise<IAddressInfo[]> {
-    const coords = await this.ethereum.getLandOf(address)
-    const info = coords.map(async coord => {
-      const data = await this.ethereum.getLandData(coord.x, coord.y)
+  async getAddressInfo(address: string): Promise<AddressInfo> {
+    const [coords, estateIds] = await Promise.all([this.ethereum.getLandOf(address), this.ethereum.getEstatesOf(address)])
 
-      return { x: coord.x, y: coord.y, name: data ? data.name : '', description: data ? data.description : '', ipns: data ? data.ipns : '' }
-    }) as Promise<IAddressInfo>[]
-    return Promise.all(info)
+    const pRequests = Promise.all(coords.map(coord => this.ethereum.getLandData(coord.x, coord.y)))
+    const eRequests = Promise.all(estateIds.map(estateId => this.ethereum.getEstateData(estateId)))
+
+    const [pData, eData] = await Promise.all([pRequests, eRequests])
+
+    const parcels =
+      pData.map((data, i) => ({
+        x: coords[i].x,
+        y: coords[i].y,
+        name: data ? data.name : '',
+        description: data ? data.description : '',
+        ipns: data ? data.ipns : ''
+      })) || []
+
+    const estates =
+      eData.map((data, i) => ({
+        id: parseInt(estateIds[i].toString(), 10),
+        name: data ? data.name : '',
+        description: data ? data.description : '',
+        ipns: data ? data.ipns : ''
+      })) || []
+
+    return { parcels, estates }
   }
 
   getWatch(): boolean {
@@ -162,7 +174,7 @@ export class Decentraland extends EventEmitter {
     return { scene, land: { ...land, owner } }
   }
 
-  async getParcelInfo(x: number, y: number): Promise<ParcelMetadata> {
+  async getParcelInfo({ x, y }: Coords): Promise<ParcelMetadata> {
     const scene = await this.localIPFS.getRemoteSceneMetadata(x, y)
     const land = await this.ethereum.getLandData(x, y)
     const owner = await this.ethereum.getLandOwner(x, y)
@@ -178,6 +190,15 @@ export class Decentraland extends EventEmitter {
     const owner = await this.ethereum.getEstateOwner(estateId)
     const parcels = await this.ethereum.getLandOfEstate(estateId)
     return { ...estate, owner, parcels }
+  }
+
+  async getEstateOfParcel(coords: Coords): Promise<Estate> {
+    const estateId = await this.ethereum.getEstateIdOfLand(coords)
+    if (!estateId) {
+      return
+    }
+
+    return this.getEstateInfo(estateId)
   }
 
   async getParcelStatus(x: number, y: number): Promise<{ lastModified?: string; files: IResolveDependency[] }> {

--- a/src/lib/Ethereum.ts
+++ b/src/lib/Ethereum.ts
@@ -23,7 +23,7 @@ factories.set('MANAToken', manaFactory)
 factories.set('LANDProxy', landFactory)
 factories.set('EstateProxy', estateFactory)
 
-export interface ILandData {
+export type LANDData = {
   version: number
   name: string
   description: string
@@ -87,10 +87,20 @@ export class Ethereum extends EventEmitter {
     return res
   }
 
-  async getLandData(x: number, y: number): Promise<ILandData> {
+  async getLandData(x: number, y: number): Promise<LANDData> {
     const contract = await Ethereum.getContract('LANDProxy')
     try {
       const landData = await contract['landData'](x, y)
+      return this.decodeLandData(landData)
+    } catch (e) {
+      fail(ErrorType.ETHEREUM_ERROR, `Unable to fetch LAND data: ${e.message}`)
+    }
+  }
+
+  async getEstateData(estateId: number): Promise<LANDData> {
+    const contract = await Ethereum.getContract('EstateProxy')
+    try {
+      const landData = await contract['getMetadata'](estateId)
       return this.decodeLandData(landData)
     } catch (e) {
       fail(ErrorType.ETHEREUM_ERROR, `Unable to fetch LAND data: ${e.message}`)
@@ -101,6 +111,15 @@ export class Ethereum extends EventEmitter {
     const contract = await Ethereum.getContract('LANDProxy')
     try {
       return await contract['ownerOfLand'](x, y)
+    } catch (e) {
+      fail(ErrorType.ETHEREUM_ERROR, `Unable to fetch LAND owner: ${e.message}`)
+    }
+  }
+
+  async getEstateOwner(estateId: number): Promise<string> {
+    const contract = await Ethereum.getContract('EstateProxy')
+    try {
+      return await contract['ownerOf'](estateId)
     } catch (e) {
       fail(ErrorType.ETHEREUM_ERROR, `Unable to fetch LAND owner: ${e.message}`)
     }
@@ -155,26 +174,7 @@ export class Ethereum extends EventEmitter {
     return landData.ipns.replace('ipns:', '')
   }
 
-  private async isLandOperator({ x, y }: ICoords, owner: string): Promise<string> {
-    const contract = await Ethereum.getContract('LANDProxy')
-    try {
-      const assetId = await contract['encodeTokenId'](x, y)
-      return await contract['isUpdateAuthorized'](owner, assetId.toString())
-    } catch (e) {
-      fail(ErrorType.ETHEREUM_ERROR, `Unable to fetch LAND authorization: ${e.message}`)
-    }
-  }
-
-  private async isEstateOperator(estateId: number, owner: string): Promise<boolean> {
-    const contract = await Ethereum.getContract('EstateProxy')
-    try {
-      return await contract['isUpdateAuthorized'](owner, estateId)
-    } catch (e) {
-      fail(ErrorType.ETHEREUM_ERROR, `Unable to fetch Estate authorization: ${e.message}`)
-    }
-  }
-
-  private async getLandOfEstate(estateId: number): Promise<ICoords[]> {
+  async getLandOfEstate(estateId: number): Promise<ICoords[]> {
     const contract = await Ethereum.getContract('EstateProxy')
     const landContract = await Ethereum.getContract('LANDProxy')
 
@@ -198,7 +198,26 @@ export class Ethereum extends EventEmitter {
     }
   }
 
-  private decodeLandData(data: string = ''): ILandData {
+  private async isLandOperator({ x, y }: ICoords, owner: string): Promise<string> {
+    const contract = await Ethereum.getContract('LANDProxy')
+    try {
+      const assetId = await contract['encodeTokenId'](x, y)
+      return await contract['isUpdateAuthorized'](owner, assetId.toString())
+    } catch (e) {
+      fail(ErrorType.ETHEREUM_ERROR, `Unable to fetch LAND authorization: ${e.message}`)
+    }
+  }
+
+  private async isEstateOperator(estateId: number, owner: string): Promise<boolean> {
+    const contract = await Ethereum.getContract('EstateProxy')
+    try {
+      return await contract['isUpdateAuthorized'](owner, estateId)
+    } catch (e) {
+      fail(ErrorType.ETHEREUM_ERROR, `Unable to fetch Estate authorization: ${e.message}`)
+    }
+  }
+
+  private decodeLandData(data: string = ''): LANDData {
     // this logic can also be found in decentraland-eth, but we can't rely on node-hid
     const version = data.charAt(0)
     switch (version) {

--- a/src/lib/Ethereum.ts
+++ b/src/lib/Ethereum.ts
@@ -5,7 +5,7 @@ import { RequestManager, ContractFactory, providers, Contract } from 'eth-connec
 
 import { isDev, getProvider } from '../utils/env'
 import { ErrorType, fail } from '../utils/errors'
-import { ICoords, isEqual, getObject } from '../utils/coordinateHelpers'
+import { Coords, isEqual, getObject } from '../utils/coordinateHelpers'
 
 import { abi as manaAbi } from '../../abi/MANAToken.json'
 import { abi as landAbi } from '../../abi/LANDRegistry.json'
@@ -24,7 +24,7 @@ factories.set('LANDProxy', landFactory)
 factories.set('EstateProxy', estateFactory)
 
 export type LANDData = {
-  version: number
+  version?: number
   name: string
   description: string
   ipns: string
@@ -75,16 +75,29 @@ export class Ethereum extends EventEmitter {
     return contract
   }
 
-  async getLandOf(address: string): Promise<ICoords[]> {
+  async getLandOf(address: string): Promise<Coords[]> {
     const contract = await Ethereum.getContract('LANDProxy')
-    let res
     try {
       const [x, y] = await contract['landOf'](address.toUpperCase())
-      res = x.map(($, i) => ({ x: $.toNumber(), y: y[i].toNumber() }))
+      return x.map(($, i) => ({ x: $.toNumber(), y: y[i].toNumber() }))
     } catch (e) {
       fail(ErrorType.ETHEREUM_ERROR, `Unable to fetch LANDs: ${e.message}`)
     }
-    return res
+  }
+
+  async getEstatesOf(address: string): Promise<number[]> {
+    const contract = await Ethereum.getContract('EstateProxy')
+    try {
+      const balance = await contract['balanceOf'](address)
+      const requests = []
+      for (let i = 0; i < balance; i++) {
+        const request = contract['tokenOfOwnerByIndex'](address, i)
+        requests.push(request)
+      }
+      return Promise.all(requests)
+    } catch (e) {
+      fail(ErrorType.ETHEREUM_ERROR, `Unable to fetch Estate IDs of owner: ${e.message}`)
+    }
   }
 
   async getLandData(x: number, y: number): Promise<LANDData> {
@@ -128,7 +141,7 @@ export class Ethereum extends EventEmitter {
   /**
    * It fails if the owner address isn't able to update given parcel (as an owner or operator)
    */
-  async validateAuthorizationOfParcel(owner: string, parcel: ICoords): Promise<void> {
+  async validateAuthorizationOfParcel(owner: string, parcel: Coords): Promise<void> {
     const isLandOperator = await this.isLandOperator(parcel, owner)
     if (!isLandOperator) {
       fail(ErrorType.ETHEREUM_ERROR, `Provided address ${owner} is not authorized to update LAND ${parcel.x},${parcel.y}`)
@@ -148,7 +161,7 @@ export class Ethereum extends EventEmitter {
   /**
    * It fails if the given parcels aren't inside the given estate
    */
-  async validateParcelsInEstate(estateId: number, parcels: ICoords[]): Promise<void> {
+  async validateParcelsInEstate(estateId: number, parcels: Coords[]): Promise<void> {
     const lands = await this.getLandOfEstate(estateId)
     const incorrectParcel = lands.find(parcel => !parcels.some(p => isEqual(parcel, p)))
     if (incorrectParcel) {
@@ -174,7 +187,7 @@ export class Ethereum extends EventEmitter {
     return landData.ipns.replace('ipns:', '')
   }
 
-  async getLandOfEstate(estateId: number): Promise<ICoords[]> {
+  async getLandOfEstate(estateId: number): Promise<Coords[]> {
     const contract = await Ethereum.getContract('EstateProxy')
     const landContract = await Ethereum.getContract('LANDProxy')
 
@@ -198,7 +211,19 @@ export class Ethereum extends EventEmitter {
     }
   }
 
-  private async isLandOperator({ x, y }: ICoords, owner: string): Promise<string> {
+  async getEstateIdOfLand({ x, y }: Coords): Promise<number> {
+    const contract = await Ethereum.getContract('EstateProxy')
+    const landContract = await Ethereum.getContract('LANDProxy')
+
+    try {
+      const assetId = await landContract['encodeTokenId'](x, y)
+      return await contract['getLandEstateId'](assetId)
+    } catch (e) {
+      fail(ErrorType.ETHEREUM_ERROR, `Unable to fetch Estate ID of LAND: ${e.message}`)
+    }
+  }
+
+  private async isLandOperator({ x, y }: Coords, owner: string): Promise<string> {
     const contract = await Ethereum.getContract('LANDProxy')
     try {
       const assetId = await contract['encodeTokenId'](x, y)

--- a/src/lib/Project.ts
+++ b/src/lib/Project.ts
@@ -89,7 +89,12 @@ export class Project {
       return this.sceneFile
     }
 
-    this.sceneFile = await readJSON<DCL.SceneMetadata>(getSceneFilePath(this.workingDir))
+    try {
+      this.sceneFile = await readJSON<DCL.SceneMetadata>(getSceneFilePath(this.workingDir))
+    } catch (e) {
+      fail(ErrorType.PROJECT_ERROR, `Unable to read 'scene.json' file. Try initializing the project using 'dcl init'`)
+    }
+
     return this.sceneFile
   }
 
@@ -296,13 +301,7 @@ export class Project {
    * Throws if a project contains an invalid main path or if the `scene.json` file is missing.
    */
   async validateExistingProject() {
-    let sceneFile
-
-    try {
-      sceneFile = await readJSON<DCL.SceneMetadata>(getSceneFilePath(this.workingDir))
-    } catch (e) {
-      fail(ErrorType.PROJECT_ERROR, `Unable to read 'scene.json' file. Try initializing the project using 'dcl init'`)
-    }
+    const sceneFile = await this.getSceneFile()
 
     if (!this.isWebSocket(sceneFile.main)) {
       if (!this.isValidMainFormat(sceneFile.main)) {

--- a/src/lib/Project.ts
+++ b/src/lib/Project.ts
@@ -16,7 +16,7 @@ import {
 } from '../utils/project'
 import ignore = require('ignore')
 import { fail, ErrorType } from '../utils/errors'
-import { inBounds, getBounds, getObject, areConnected, ICoords } from '../utils/coordinateHelpers'
+import { inBounds, getBounds, getObject, areConnected, Coords } from '../utils/coordinateHelpers'
 
 export enum BoilerplateType {
   TYPESCRIPT_STATIC = 'ts-static',
@@ -219,7 +219,7 @@ export class Project {
   /**
    * Returns a promise of an object containing the base X and Y coordinates for a parcel.
    */
-  async getParcelCoordinates(): Promise<ICoords> {
+  async getParcelCoordinates(): Promise<Coords> {
     const sceneFile = await this.getSceneFile()
     const { base } = sceneFile.scene
     return getObject(base)
@@ -228,7 +228,7 @@ export class Project {
   /**
    * Returns a promise of an array of the parcels of the scene
    */
-  async getParcels(): Promise<ICoords[]> {
+  async getParcels(): Promise<Coords[]> {
     const sceneFile = await this.getSceneFile()
     return sceneFile.scene.parcels.map(getObject)
   }

--- a/src/utils/coordinateHelpers.ts
+++ b/src/utils/coordinateHelpers.ts
@@ -5,7 +5,7 @@ export interface IBounds {
   maxY: number
 }
 
-export type ICoords = {
+export type Coords = {
   x: number
   y: number
 }
@@ -84,21 +84,21 @@ export function isValid(val: string): boolean {
  * Converts a string-based set of coordinates to an object
  * @param coords A string containing a set of coordinates
  */
-export function getObject(coords: string): ICoords
+export function getObject(coords: string): Coords
 /**
  * Converts a array-based set of coordinates to an object
  * @param coords An array containing a set of coordinates
  */
-export function getObject(coords: number[]): ICoords
-export function getObject(coords: number[] | string): ICoords {
+export function getObject(coords: number[]): Coords
+export function getObject(coords: number[] | string): Coords {
   const [x, y] = typeof coords === 'string' ? parse(coords)[0].split(',') : coords
   return { x: parseInt(x.toString(), 10), y: parseInt(y.toString(), 10) }
 }
 
 /**
- * Converts a ICoords object to a string-based set of coordinates
+ * Converts a Coords object to a string-based set of coordinates
  */
-export function getString({ x, y }: ICoords): string {
+export function getString({ x, y }: Coords): string {
   return `${x},${y}`
 }
 
@@ -113,7 +113,7 @@ export function inBounds(x: number, y: number): boolean {
 /**
  * Returns true if the given parcels array are connected
  */
-export function areConnected(parcels: ICoords[]): boolean {
+export function areConnected(parcels: Coords[]): boolean {
   if (parcels.length === 0) {
     return false
   }
@@ -121,7 +121,7 @@ export function areConnected(parcels: ICoords[]): boolean {
   return visited.length === parcels.length
 }
 
-function visitParcel(parcel: ICoords, allParcels: ICoords[] = [parcel], visited: ICoords[] = []): ICoords[] {
+function visitParcel(parcel: Coords, allParcels: Coords[] = [parcel], visited: Coords[] = []): Coords[] {
   let isVisited = visited.some(visitedParcel => isEqual(visitedParcel, parcel))
   if (!isVisited) {
     visited.push(parcel)
@@ -132,14 +132,14 @@ function visitParcel(parcel: ICoords, allParcels: ICoords[] = [parcel], visited:
 }
 
 function getIsNeighbourMatcher(x: number, y: number) {
-  return (coords: ICoords) =>
+  return (coords: Coords) =>
     (coords.x === x && (coords.y + 1 === y || coords.y - 1 === y)) || (coords.y === y && (coords.x + 1 === x || coords.x - 1 === x))
 }
 
-function getNeighbours(x: number, y: number, parcels: ICoords[]): ICoords[] {
+function getNeighbours(x: number, y: number, parcels: Coords[]): Coords[] {
   return parcels.filter(getIsNeighbourMatcher(x, y))
 }
 
-export function isEqual(p1: ICoords, p2: ICoords): boolean {
+export function isEqual(p1: Coords, p2: Coords): boolean {
   return p1.x === p2.x && p1.y === p2.y
 }

--- a/src/utils/coordinateHelpers.ts
+++ b/src/utils/coordinateHelpers.ts
@@ -5,7 +5,7 @@ export interface IBounds {
   maxY: number
 }
 
-export interface ICoords {
+export type ICoords = {
   x: number
   y: number
 }

